test/e2e/syncer/dns/dns_test.go

```diff
@@ -106,6 +119,41 @@ func TestDNSResolution(t *testing.T) {
 	t.Log("Checking DNS name does not resolve across workspaces")
 	framework.Eventually(t, checkLogs(ctx, t, downstreamKubeClient, downstreamWS2NS1, "ping-fully-qualified-fail", "ping: bad"),
 		wait.ForeverTestTimeout, time.Millisecond*500, "Service name was resolved")
+
+	t.Log("Change ping-fully-qualified deployment DNS config to use workspace 2 nameserver")
+	dnsServices, err := downstreamKubeClient.CoreV1().Services(syncer.SyncerID).List(ctx, metav1.ListOptions{})
+	require.NoError(t, err)
+	require.True(t, len(dnsServices.Items) >= 2)
+
+	deployment, err := downstreamKubeClient.AppsV1().Deployments(downstreamWS1NS1).Get(ctx, "ping-fully-qualified", metav1.GetOptions{})
+	require.NoError(t, err)
+
+	existingDNSIP := deployment.Spec.Template.Spec.DNSConfig.Nameservers[0]
+	newDNSIP := ""
+	for _, svc := range dnsServices.Items {
+		if strings.HasPrefix(svc.Name, "kcp-dns-") {
+			if svc.Spec.ClusterIP != existingDNSIP {
+				newDNSIP = svc.Spec.ClusterIP
+				break
+			}
+		}
+	}
+	require.NotEmpty(t, newDNSIP, "could not find another DNS service")
+	deployment.Spec.Template.Spec.DNSConfig.Nameservers[0] = newDNSIP
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		deployment, err := downstreamKubeClient.AppsV1().Deployments(downstreamWS1NS1).Get(ctx, "ping-fully-qualified", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		deployment.Spec.Template.Spec.DNSConfig.Nameservers[0] = newDNSIP
+		_, err = downstreamKubeClient.AppsV1().Deployments(downstreamWS1NS1).Update(ctx, deployment, metav1.UpdateOptions{})
+		return err
+	})
+	require.NoError(t, err)
+
+	framework.Eventually(t, checkLogs(ctx, t, downstreamKubeClient, downstreamWS1NS1, "ping-fully-qualified", "ping: bad"),
+		wait.ForeverTestTimeout, time.Millisecond*500, "Service name was still not resolved")
```

Comment:
I don't feel the error message appropriate here. What do we test ? Isn't it the fact that deployment should *not* be able to access to the DNS server of workspace 2  ? Shouldn't this be reflected in the error message ?